packages/sqflite/tizen/src/sqflite_plugin.cc

```diff
@@ -0,0 +1,796 @@
+        database_map_.erase(*existing_database_id);
+        single_instances_by_path_.erase(path);
+      }
+    }
+    // TODO: Safe check before delete.
+    std::filesystem::remove(path);
+    result->Success();
+  }
+
+  flutter::EncodableValue MakeOpenResult(int database_id,
+                                         bool recovered_in_transaction) {
+    flutter::EncodableMap response;
+    response.insert(
+        std::make_pair(flutter::EncodableValue(sqflite_constants::kParamId),
+                       flutter::EncodableValue(database_id)));
+    if (recovered_in_transaction) {
+      response.insert(
+          std::make_pair(flutter::EncodableValue(
+                             sqflite_constants::kParamRecoveredInTransaction),
+                         flutter::EncodableValue(true)));
+    }
+    return flutter::EncodableValue(response);
+  }
+
+  void OnOpenDatabaseCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    flutter::EncodableMap arguments =
+        std::get<flutter::EncodableMap>(*method_call.arguments());
+    std::string path;
+    bool read_only = false;
+    bool single_instance = false;
+
+    GetValueFromEncodableMap(arguments, sqflite_constants::kParamPath, path);
+    GetValueFromEncodableMap(arguments, sqflite_constants::kParamReadOnly,
+                             read_only);
+    GetValueFromEncodableMap(arguments, sqflite_constants::kParamSingleInstance,
+                             single_instance);
+
+    const bool in_memory = IsInMemoryPath(path);
+    single_instance = single_instance && !in_memory;
+
+    if (single_instance) {
+      int found_database_id = 0;
+      auto sit = single_instances_by_path_.find(path);
+      if (sit != single_instances_by_path_.end()) {
+        found_database_id = sit->second;
+      }
+      if (found_database_id) {
+        auto dit = database_map_.find(found_database_id);
+        if (dit != database_map_.end()) {
+          if (dit->second->database()) {
+            auto response = MakeOpenResult(found_database_id, true);
+            result->Success(response);
+            return;
+          }
+        }
+      }
+    }
+    // TODO: Protect with mutex
+    const int new_database_id = ++database_id_;
+    try {
+      std::shared_ptr<sqflite_database::DatabaseManager> database_manager =
+          std::make_shared<sqflite_database::DatabaseManager>(
+              path, new_database_id, single_instance, 0);
+      if (!read_only) {
+        LOG_DEBUG("opening read-write database in path %s", path.c_str());
+        database_manager->Open();
+      } else {
+        LOG_DEBUG("opening read only database in path %s", path.c_str());
+        database_manager->OpenReadOnly();
+      }
+
+      // Store dbid in internal map
+      // TODO: Protect with mutex
+      LOG_DEBUG("saving database id %d for path %s", database_id_,
```

Comment:
Why don't you use `new_database_id` instead of `database_id_`? 
(include 563, 567, 575 line)